src/jobservice/job/impl/gc/garbage_collection.go

```diff
@@ -139,29 +158,21 @@ func (gc *GarbageCollector) init(ctx job.Context, params job.Parameters) error {
 	gc.redisURL = params["redis_url_reg"].(string)
 
 	// default is to delete the untagged artifact
-	if params["delete_untagged"] == "" {
+	deleteUntagged, exist := params["delete_untagged"]
```

Comment:
I think you just need a convention `ok` flag

```go
deleteUntagged, exist := params["delete_untagged"]
if exist {
   if untagged, ok := deleteUntagged.(bool); ok && untagged {
      // do untagged things
  }
}
```

Reply:
updated